objects/src/mock/chain.rs

```diff
@@ -55,3 +55,248 @@ pub fn mock_chain_data(consumed_notes: &mut [Note]) -> ChainMmr {
+            let account = AccountBuilder::new(rng)
+                .add_storage_items(storage)
+                .add_assets(assets)
+                .account_type(account_type)
+                .on_chain(on_chain)
+                .code(code)
+                .build()?;
+            let account_id = account.id();
+            self.pending_objects.accounts.push(account);
+            Ok(account_id)
+        }
+
+        /// Creates a [AccountId] with type [AccountType::FungibleFaucet] and add to the list of
+        /// pending objects.
+        pub fn new_fungible_faucet(
+            &mut self,
+            on_chain: bool,
+            code_root: Digest,
+            storage_root: Digest,
+        ) -> Result<AccountId, AccountError> {
+            let faucet_id = self
+                .account_id_builder
+                .account_type(AccountType::NonFungibleFaucet)
+                .on_chain(on_chain)
+                .code_root(code_root)
+                .storage_root(storage_root)
+                .build()?;
+            let builder = FungibleAssetBuilder::new(faucet_id)
+                .expect("builder was not configured to create fungible faucets");
+            self.pending_objects.fungible_faucets.push((faucet_id, builder));
+            Ok(faucet_id)
+        }
+
+        /// Creates a [AccountId] with type [AccountType::NonFungibleFaucet] and add to the list of
+        /// pending objects.
+        pub fn new_nonfungible_account(
+            &mut self,
+            on_chain: bool,
+            code_root: Digest,
+            storage_root: Digest,
+        ) -> Result<AccountId, AccountError> {
+            let faucet_id = self
+                .account_id_builder
+                .account_type(AccountType::NonFungibleFaucet)
+                .on_chain(on_chain)
+                .code_root(code_root)
+                .storage_root(storage_root)
+                .build()?;
+            let rng = R::from_rng(&mut self.rng).expect("rng seeding failed");
+            let builder = NonFungibleAssetBuilder::new(faucet_id, rng)
+                .expect("builder was not configured to build nonfungible faucets");
+            self.pending_objects.nonfungible_faucets.push((faucet_id, builder));
+            Ok(faucet_id)
+        }
+
+        /// Creates [FungibleAsset] from the fungible faucet at position `asset_pos`.
+        pub fn new_fungible_asset(
+            &mut self,
+            asset_pos: usize,
+            amount: u64,
+        ) -> Result<FungibleAsset, AssetError> {
+            self.objects.fungible_faucets[asset_pos].1.amount(amount)?.build()
+        }
+
+        /// Creates [NonFungibleAsset] from the nonfungible faucet at position `asset_pos`.
+        pub fn new_nonfungible_asset(
+            &mut self,
+            asset_pos: usize,
+        ) -> Result<NonFungibleAsset, AssetError> {
+            self.objects.nonfungible_faucets[asset_pos].1.build()
+        }
```

Comment:
nit: should `asset_pos` actually be `faucet_pos`?

Also, is there a reason not to return just `Asset` from both of this functions?